bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,322 @@
+ * sent to one of the channels.
+ *
+ * @author Jeremy Stucki - Initial contribution
+ */
+@NonNullByDefault
+public class PublicTransportSwitzerlandStationboardHandler extends BaseThingHandler {
+
+    // Limit the API response to the necessary fields
+    private static final String FIELD_FILTERS = createFilterForFields(
+            "stationboard/to",
+            "stationboard/category",
+            "stationboard/number",
+            "stationboard/stop/departureTimestamp",
+            "stationboard/stop/delay",
+            "stationboard/stop/platform");
+
+    private final ChannelGroupUID dynamicChannelGroupUID = new ChannelGroupUID(getThing().getUID(), "departures");
+
+    private final Logger logger = LoggerFactory.getLogger(PublicTransportSwitzerlandStationboardHandler.class);
+
+    private final SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm");
+
+    private @Nullable ScheduledFuture<?> updateChannelsJob;
+    private @Nullable ExpiringCache<@Nullable JsonElement> cache;
+    private @Nullable PublicTransportSwitzerlandStationboardConfiguration configuration;
+
+    public PublicTransportSwitzerlandStationboardHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannels();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Together with the 10 second timeout, this should be less than a minute
+        cache = new ExpiringCache<>(45_000, this::updateData);
+
+        PublicTransportSwitzerlandStationboardConfiguration configuration = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+        this.configuration = configuration;
+
+        @Nullable String configurationError = findConfigurationError(configuration);
+        if (configurationError != null) {
+            stopChannelUpdate();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, configurationError);
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            startChannelUpdate();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopChannelUpdate();
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+
+        PublicTransportSwitzerlandStationboardConfiguration configuration = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+        this.configuration = configuration;
+
+        @Nullable String configurationError = findConfigurationError(configuration);
+        if (configurationError != null) {
+            stopChannelUpdate();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, configurationError);
+        } else if (updateChannelsJob == null || updateChannelsJob.isCancelled()) {
```

Comment:
You could fix the compiler warning by storing `updateChannelsJob` to a local variable and do the null check on that.